[Modules] Format the metadata.json and aligned the Name and Description for all Modules and Updated Readme (#3309)
Azure/ResourceModules: modules/network/network-managers/security-admin-configurations/rule-collections/rules/README.md

> # AVNM Security Admin Configuration Rule Collection Rules `[Microsoft.Network/networkManagers/securityAdminConfigurations/ruleCollections/rules]`

Reviewer: ```suggestion
# Network Manager Security Admin Configuration Rule Collection Rules `[Microsoft.Network/networkManagers/securityAdminConfigurations/ruleCollections/rules]`
```

> This module deploys an Azure Virtual Network Manager (AVNM) Security Admin Configuration Rule Collection Rule.

Reviewer: ```suggestion
This module deploys a Network Manager Security Admin Configuration Rule Collection Rule.
```
